test/variational/test_exact_state.py

```diff
@@ -0,0 +1,242 @@
+# Copyright 2021 The NetKet Authors - All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+
+import pytest
+from pytest import approx, raises, warns
+
+import numpy as np
+import jax
+import netket as nk
+from jax.nn.initializers import normal
+
+from .. import common
+
+pytestmark = common.skipif_mpi
+
```

Comment:
Also woul dbe good to add a test for the (exact) qgt against computing it by hand.